src/Property.php

```diff
@@ -67,7 +67,7 @@ protected function resolveTypeDefinition()
             return;
         }
 
-        preg_match('/\@var ([\w|\\\\]+)/', $docComment, $matches);
+        preg_match('/\@var ([\w|\\\\\[\]]+)/', $docComment, $matches);
```

Comment:
I think we'd better change this regex slightly:

```
/\@var ([\w|\\\\]+(\[\])?)/
```

This way `[]` is required to be together.

Reply:
This will add another group (so `$varDocComment = end($matches);` won't work) and it doesn't allow for multiple types using the pipe character `|`.

I'll try and get a regex pattern together that solves this.